### PR TITLE
adding ptrace to asan

### PR DESCRIPTION
**Pull request by billfeng327**
ASAN supports PTRACE only in docker privileged mode, adding ptrace and docker privileged flag.

### CODE REVIEW

File: tools/internal_ci/linux/grpc_run_tests_matrix.sh [hunk 1]
```diff
@@ -23,7 +23,13 @@ source tools/internal_ci/helper_scripts/prepare_build_linux_rc
 # If this is a PR using RUN_TESTS_FLAGS var, then add flags to filter tests
 if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ] && [ -n "$RUN_TESTS_FLAGS" ]; then
   sudo apt-get update
-  sudo apt-get install -y jq
+  retry=0
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_run_tests_matrix.sh]: this is not the only place where jq gets installed (in fact I'm surprised what is this snippet doing in this script).

let's do this unrelated change in a separate PR, it doesn't really fit under the "adding ptrace to asan" umbrella.

**Comment by jtattermusch** [comment]
What is the motivation for this PR? Is it just cleanup or are we fixing something that's currently broken (or could be broken soon)?

**Comment by jtattermusch** [comment]
As this change is purely about bazel RBE and independent from run_tests_matrix.py independent, let's do one of these:
1. fix https://github.com/grpc/grpc/issues/17794 first (in a separate PR) and then merge this PR once tests are green.
2. remove the jq fix from this PR, mark known flakes as #17794 (the bazel RBE tests which we care about shouldn't be affected by the flake) and merge.

**Comment by jtattermusch** [comment]
Known failures (adding them myself to be able to merge and hopefully fix https://github.com/grpc/grpc/issues/17846.

- https://github.com/grpc/grpc/issues/17850
- https://github.com/grpc/grpc/issues/17851
- https://github.com/grpc/grpc/issues/17847

File: third_party/toolchains/BUILD [hunk 2]
```diff
@@ -56,6 +56,14 @@ platform(
             name: "dockerNetwork"
             value: "off"
         }
+        properties: {
```
Comment (jtattermusch) [inline third_party/toolchains/BUILD]: Looks like this is being set for all our RBE tests, not just ASAN?

Comment (billfeng327) [reply third_party/toolchains/BUILD]: yes, Igor Foox mentioned that it should be applied to all RBE builds
